usr/share/rear/finalize/GNU/Linux/320_migrate_network_configuration_files.sh

```diff
@@ -1,155 +1,493 @@
+                Log "Wrote new MAC addresses and network interfaces in $network_config_file"
+            else
+                LogPrintError "Failed to rewrite MAC addresses and network interfaces in $network_config_file"
+            fi
+        done
+    else
+        Log "No rewriting of MAC addresses and network interfaces (empty sed_script)"
+    fi
+    # Rename network configuration files where the file name contains the MAC address or the interface name:
+    for network_config_file in "${network_config_files[@]}" ; do
+        # E.g. when the interface has changed from eth0 to eth1 the sed_script contains "... ; s/eth0/eth1/g" (cf. "Get new interface" above)
+        # so when this sed_script is applied to a network configuration file name like $TARGET_FS_ROOT/etc/sysconfig/network/ifcfg-eth0
+        # the new_file_name becomes $TARGET_FS_ROOT/etc/sysconfig/network/ifcfg-eth1
+        new_file_name="$( sed -e "$sed_script" <<<"$network_config_file" )"
+        if test "$new_file_name" -a "$network_config_file" != "$new_file_name" ; then
+            Log "Renaming '$network_config_file' as '$new_file_name'"
+            mv $v "$network_config_file" "$new_file_name" || LogPrintError "Failed to rename '$network_config_file' as '$new_file_name'"
         fi
-    done < <( read_and_strip_file $CONFIG_DIR/mappings/mac | sed -e 'p;y/abcdef/ABCDEF/' )
-    #                                              ^^^^^^^
-    #       this is a nasty hack that prints each line as is (lowercase) and once again in uppercase
-    #       the reason is that the mac mappings are in lower case but some systems seem to keep
-    #       the MAC adresses in upper case and since I don't want to mess with this I treat it as
-    #       separate things and do the replacement case-sensitive
-
-    Debug "sed_script: '$sed_script'"
-
-    for patch_file in "${PATCH_FILES[@]}" ; do
-        sed -i -e "$sed_script" "$patch_file" || LogPrintError "Migrating network configuration in $patch_file failed"
     done
-
-    # rename files
-    for file in "${PATCH_FILES[@]}"; do
-        new_file="$(sed -e "$sed_script" <<<"$file")"
-        if test "$new_file" -a "$new_file" != "$file" ; then
-            mv $v "$file" "$new_file" >&2
+else
+    # When .../mappings/ip_addresses or .../mappings/routes exists but .../mappings/mac is missing or has no content
+    # we need a .../mappings/mac file because otherwise the logic to rewrite IP addresses or routes would fail.
+    # We try to generate one from .../mappings/ip_addresses with old_mac=new_mac for non-migrated interfaces:
+    if test -s $TMP_DIR/mappings/ip_addresses ; then
+        for interface in $( cut -f 1 -d " " $TMP_DIR/mappings/ip_addresses ) ; do
+            # /sys/class/net/$interface/address contains the MAC address with lower case hex letters (cf. above):
+            current_mac=$( cat /sys/class/net/$interface/address )
+            echo "$mac $mac $interface" >> $TMP_DIR/mappings/mac
```

Comment:
@yontalcar 
thank you so much for your careful review!

This is clearly a leftover from the old code that was
```
# for the unlikely case where an ip address or routes mapping is supplied but a mac
# mapping is missing, we have to fake a mac mapping file. Otherwise the
# logic to rewrite the ip addresses or routes will fail.
# The easiest way is to be sure to always provide a mac mapping files with "old_mac new_mac interface" format
# with old_mac=new_mac for non-migrated interfaces.

if ! test -s $TMP_DIR/mappings/mac ; then
    for interface in $(cut -f 1 -d " " $TMP_DIR/mappings/ip_addresses) ; do
        mac=$(cat /sys/class/net/$interface/address)
        echo "$mac $mac $interface" >> $TMP_DIR/mappings/mac
    done
fi
```